framework/src/modules/nft/method.ts

```diff
@@ -278,21 +278,29 @@ export class NFTMethod extends BaseMethod {
 	public async getNextAvailableIndex(
 		methodContext: MethodContext,
 		collectionID: Buffer,
-	): Promise<number> {
+	): Promise<bigint> {
+		const indexLength = LENGTH_NFT_ID - LENGTH_CHAIN_ID - LENGTH_COLLECTION_ID;
 		const nftStore = this.stores.get(NFTStore);
+
 		const nftStoreData = await nftStore.iterate(methodContext, {
-			gte: Buffer.alloc(LENGTH_NFT_ID, 0),
-			lte: Buffer.alloc(LENGTH_NFT_ID, 255),
+			gte: Buffer.concat([this._config.ownChainID, collectionID, Buffer.alloc(indexLength, 0)]),
+			lte: Buffer.concat([this._config.ownChainID, collectionID, Buffer.alloc(indexLength, 255)]),
 		});
 
-		let count = 0;
-		for (const { key } of nftStoreData) {
-			if (key.slice(LENGTH_CHAIN_ID, LENGTH_CHAIN_ID + LENGTH_COLLECTION_ID).equals(collectionID)) {
-				count += 1;
-			}
+		if (nftStoreData.length === 0) {
+			return BigInt(0);
+		}
+
+		const latestKey = nftStoreData[nftStoreData.length - 1].key;
+		const indexBytes = latestKey.slice(LENGTH_CHAIN_ID + LENGTH_COLLECTION_ID, LENGTH_NFT_ID);
+		const index = indexBytes.readBigUInt64BE();
+		const largestIndex = BigInt(BigInt(2 ** 64) - BigInt(1));
+
+		if (index === largestIndex) {
+			throw new Error('No more available indexes');
 		}
 
-		return count;
+		return index + BigInt(1);
```

Comment:
Apparently the [LIP PR](https://github.com/LiskHQ/lips/pull/412) was updated after this change. I think it needs to be updated to align per the [latest commit](https://github.com/LiskHQ/lips/pull/412/commits/944a1e71c4280f0001f1b84e7e3c58ba86f990eb)

Reply:
I believe that now the LIP is closer to implementation than before, @has5aan please comment

Reply:
On second thought: yes, it's closer regarding the store iteration, but the last part of the pseudocode has slightly changed

Reply:
Looking more closely, I now see that the overall logic is still the same after the [commit](https://github.com/LiskHQ/lips/pull/412/commits/944a1e71c4280f0001f1b84e7e3c58ba86f990eb) but was only a syntactical change. I resolve my comment.